pkg/controllers/state/cluster.go

```diff
@@ -293,6 +296,37 @@ func (c *Cluster) Reset() {
 	c.nameToProviderID = map[string]string{}
 	c.bindings = map[types.NamespacedName]string{}
 	c.antiAffinityPods = sync.Map{}
+	c.daemonSetPods = map[types.NamespacedName]*v1.Pod{}
+}
+
+func (c *Cluster) GetDaemonSetPods(daemonset *appsv1.DaemonSet) *v1.Pod {
+	c.mu.RLock()
```

Comment:
You have to be careful here. This `RLock` only works until the end of this func but you are returning a pointer, which means that you will be accessing this data outside of this func.